contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -674,11 +692,47 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
         );
     }
 
+    /**
+     * Gets the queue element at a particular index.
+     * @param _index Index of the queue element to access.
+     * @return _element Queue element at the given index.
+     */
+    function _getQueueElement(
+        uint256 _index,
+        iOVM_ChainStorageContainer _queueRef
+    )
+        internal
+        view
+        returns (
+            Lib_OVMCodec.QueueElement memory _element
+        )
+    {
+        (
+            bytes32 queueRoot,
+            bytes32 timestampAndBlockNumber
+        ) = _queueRef.get2(uint40(_index * 2));
```

Comment:
Make use of `get2` to reduce number of `STATICCALL`s.

Reply:
🔥 `get2` ftw

Reply:
I think we should add a comment clarifying why the index is doubled.